src/inmanta/db/versions/v202402130.py

```diff
@@ -0,0 +1,51 @@
+"""
+    Copyright 2024 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+
+from asyncpg import Connection
+
+
+async def update(connection: Connection) -> None:
+    schema = """
+    UPDATE public.configurationmodel
+    SET undeployable=(
+        CASE
+            WHEN undeployable IS NULL
+            -- Set default to empty array
+            THEN ARRAY[]::character varying[]
```

Comment:
If I read this correctly, this contains first a constructor call, then a type cast and them a type definition. Do you know why this is required? The type of the `undeployable` is already known by PostgreSQL, no?

Reply:
Right, I'll simplify this

Reply:
I don't think I see what you mean here:
> and them a type definition 

To me it is only a constructor `ARRAY[]` into a typecast of type array of `character varying`

And we cannot use only the constructor or else we get :
```
cannot determine type of empty array
               HINT:  Explicitly cast to the desired type, for example ARRAY[]::integer[].
```

Reply:
I'll go back to my inital implementation of `THEN '{}'`

Reply:
I would expect the following:
```suggestion
            THEN ARRAY[]::varying[]
```
So that you have:

* `ARRAY[]` that creates a new array
* `::varying[]` which type casts the array to the correct type.

But the construct `ARRAY[]::character varying[]` I am unable to interpret. That doesn't mean it's incorrect of course. So that's why I am asking why it was done like this and how I should read it. It would also help if you could point me to the relevant PostgreSQL documentation for example.

I see you now changed it into `'{}'`. That's fine by me as well.

Reply:
I think I see,  `character varying` is an actual type itself (https://www.postgresql.org/docs/current/datatype-character.html#DATATYPE-CHARACTER) so these are equivalent :

```sql
THEN ARRAY[]::varchar[]
```

And
```sql
THEN ARRAY[]::character varying[]
 ```